Ok, so I'm pretty sure whatever these fails are must be unrelated -- they occur on both main and this PR, and only started popping up tonight. Fails only seem to affect these `test_name()` tests and similar... not sure if I upgraded a dependency but this is happening in a clean `pip install -e .[all,dev]` 😢 
```
__________________________________ test_name __________________________________

    def test_name():
        """Test setting layer name."""
        np.random.seed(0)
        data = np.random.randint(20, size=(10, 15))
        layer = Labels(data)
>       assert layer.name == 'Labels'
E       AssertionError: assert 'data' == 'Labels'
E         
E         - Labels
E         + data
```

--- a/napari/conftest.py
+++ b/napari/conftest.py
@@ -950,7 +950,18 @@ with contextlib.suppress(ImportError):
 
     @pytest.fixture(scope='session')
     def qapp_cls():
+        """The qapp fixture uses the qapp_cls fixture to select
+        the class to use for create the QApplication instance.
+
+        As qapp fixture is using more complex logic, we decided
+        not to override it but overwrite the fixture used by it.
+
+        We need to set attributte before the QApplication is created.
+        """
         if PYQT5 or PYSIDE2:
+            # As Qt6 autodetect High dpi scaling, we need to
+            # enable it only on Qt5 bindings.
+            # https://doc.qt.io/qtforpython-6/faq/porting_from2.html#class-function-deprecations
             QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)
         return QApplication
 

